src/main/scala/cognite/spark/v1/wdl/WellDataLayerRDD.scala

```diff
@@ -0,0 +1,119 @@
+package cognite.spark.v1.wdl
+
+import cats.effect.IO
+import cognite.spark.v1.{CdfPartition, CdfSparkException, CdpConnector, RelationConfig}
+import com.cognite.sdk.scala.common.ItemsWithCursor
+import com.cognite.sdk.scala.v1.GenericClient
+import io.circe.JsonObject
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types._
+import org.apache.spark.{Partition, SparkContext, TaskContext}
+
+import scala.collection.AbstractIterator
+
+class WellDataLayerRDD(
+    @transient override val sparkContext: SparkContext,
+    val schema: StructType,
+    val model: String,
+    val config: RelationConfig
+) extends RDD[Row](sparkContext, Nil) {
+  import CdpConnector._
+
+  @transient lazy val client: GenericClient[IO] =
+    CdpConnector.clientFromConfig(config)
+
+  // scalastyle:off cyclomatic.complexity
+  private def getReadUrlPart(modelType: String): String =
+    modelType.replace("Ingestion", "") match {
+      case "Well" => "wells/list"
+      case "DepthMeasurement" => "measurements/depth/list"
+      case "TimeMeasurement" => "measurements/time/list"
+      case "RigOperation" => "rigoperations/list"
+      case "HoleSectionGroup" => "holesections/list"
+      case "WellTopGroup" => "welltops/list"
+      case "Npt" => "npt/list"
+      case "Nds" => "npt/list"
+      case "CasingSchematic" => "casings/list"
+      case "Trajectory" => "trajectories/list"
+      case _ => sys.error(s"Unknown model type: $modelType")
+    }
+  // scalastyle:on cyclomatic.complexity
```

Comment:
Maybe use a local Map instead of this long list of pattern match to avoid cyclomatic.complexity?
Smthing like 
```scala
private val modelTypeToUrl = Map("Well" -> "wells/list",  "DepthMeasurement" -> "measurements/depth/list" , ...)
private def getReadUrlPart(modelType: String): String = {
    val key = modelType.replace("Ingestion", "")
    modelTypeToUrl.get(key).getOrElse(throw new CdfSparkException..)
}
```

Reply:
Should be the same for modern compilers (I hope), but this old-school way is definitely good! :)